### PR TITLE
= actor add clarify message to dead letter listener

### PR DESCRIPTION
**Pull request by He-Pin**
refs https://github.com/akka/akka/issues/24875

### CODE REVIEW

File: akka-actor/src/main/scala/akka/event/DeadLetterListener.scala [hunk 1]
```diff
@@ -33,9 +33,16 @@ class DeadLetterListener extends Actor {
       val done = maxCount != Int.MaxValue && count >= maxCount
       val doneMsg = if (done) ", no more dead letters will be logged" else ""
       eventStream.publish(Info(rcp.path.toString, rcp.getClass,
-        s"Message [${message.getClass.getName}] $origin to $rcp was not delivered. [$count] dead letters encountered$doneMsg. " +
-          "This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' " +
-          "and 'akka.log-dead-letters-during-shutdown'."))
+        s"""
+           |Message [${message.getClass.getName}] from [$origin] to [$rcp] was not delivered.
+           |[$count] dead letters encountered $doneMsg.
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/event/DeadLetterListener.scala]: no space before doneMsg, formatting is wrong currently here for the "" case

File: akka-actor/src/main/scala/akka/event/DeadLetterListener.scala [hunk 2]
```diff
@@ -33,9 +33,16 @@ class DeadLetterListener extends Actor {
...
+           |
+           |If this is not an expected behavior, then the [$rcp] may died unexpectedly,
+           |You can inspect the cause by adding supervision strategy to it's parent.
+           |
+           |This logging can be turned off or adjusted with configuration settings:
+           |'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
+         """.stripMargin))
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/event/DeadLetterListener.scala]: -1 for line breaks in log messages

File: akka-actor/src/main/scala/akka/event/DeadLetterListener.scala [hunk 3]
```diff
@@ -33,9 +33,16 @@ class DeadLetterListener extends Actor {
...
       val doneMsg = if (done) ", no more dead letters will be logged" else ""
       eventStream.publish(Info(rcp.path.toString, rcp.getClass,
-        s"Message [${message.getClass.getName}] $origin to $rcp was not delivered. [$count] dead letters encountered$doneMsg. " +
-          "This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' " +
-          "and 'akka.log-dead-letters-during-shutdown'."))
+        s"""
+           |Message [${message.getClass.getName}] from [$origin] to [$rcp] was not delivered.
+           |[$count] dead letters encountered $doneMsg.
+           |
+           |If this is not an expected behavior, then the [$rcp] may died unexpectedly,
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/event/DeadLetterListener.scala]: died => terminated

Comment (ktoso) [reply akka-actor/src/main/scala/akka/event/DeadLetterListener.scala]: remove " the "?

**Review by ktoso — COMMENTED** [review]
Good improvement but needs some fixes, I commented in line

**Comment by johanandren** [comment]
If we are going to have this long log entry, which I'm not sure I agree upon, I think it should at least not be logged every time but maybe only the first time there is a dead letter or something along those lines.

File: akka-actor/src/main/scala/akka/event/DeadLetterListener.scala [hunk 4]
```diff
@@ -33,9 +33,16 @@ class DeadLetterListener extends Actor {
...
+           |You can inspect the cause by adding supervision strategy to it's parent.
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/event/DeadLetterListener.scala]: I'm not sure this advice will help many. Default supervision for exceptions is to restart the actor and then there will be no deadLetters, and if the actor is stopped the parent supervision will not really see that unless it's watching.

**Review by patriknw — COMMENTED** [review]
agree with Konrad's comments, and I'm also skeptical to the actual advice.

The formatting with `[]` is good, though :-)

**Comment by johanandren** [comment]
Any plans of pushing this forward @hepin1989 ?

**Comment by hepin1989** [comment]
Will update later ,I was busy on the book these days, so sorry.

**Comment by ktoso** [comment]
I fixed a past tense wording issue and will merge once validates https://github.com/akka/akka/pull/24941/commits/6b204ff1211c68ebaf9a15e6582bfa8495712e0b

Thanks a lot @hepin1989

**Review by ktoso — APPROVED** [review]

**Review by raboof — APPROVED** [review]
OK :+1:

**Pull request merged**